sycl/test/esimd/on-device/BitonicSortK.cpp

```diff
@@ -5,9 +5,8 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
-// TODO enable on Windows
-// REQUIRES: linux
-// REQUIRES: gpu
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux,gpu,opencl
```

Comment:
I think opencl feature is enabled only if Level Zero driver is not installed. https://github.com/intel/llvm/blob/sycl/sycl/test/lit.cfg.py#L196

So this change seems to disable ESIMD test validation in our CI, where machines have Level Zero device available.

Something like this might keep test coverage in CI and disable these tests on the systems with CUDA devices:
```suggestion
// REQUIRES: linux && gpu
// UNSUPPORTED: cuda
```

Reply:
```cpp
// REQUIRES: linux && gpu
// UNSUPPORTED: cuda
```
would allow to execute on LevelZero plugin where ESIMD is not supported. Is there a way to disable LevelZero?

Reply:
@vladimirlaz , can you advise on the above?

Reply:
shouldn't that work?

// REQUIRES: linux && gpu
// UNSUPPORTED: cuda || level_zero

Reply:
Looks like 'UNSUPPORTED: level_zero' will disable execution on machines where Level Zero is installed, right? So this won't work. The variant @bader suggested should actually be OK - ESIMD tests force SYCL_BE=PI_OPENCL for now, so I'll use it. Thanks.

Reply:
Please note that target BE is set by LIT parameter in CMkaeFiles.txt. So we always have 1 target BE selected in LIT infra.
There should be no issue
```
user@host:/WS/build$ bin/llvm-lit -v ../sycl/test/esimd/on-device/BitonicSortK.cpp --param SYCL_BE=PI_LEVEL_ZERO
llvm-lit: /WS/build/bin/../../llvm/utils/lit/lit/llvm/config.py:347: note: using clang: /WS/build/bin/clang
llvm-lit: /WS/sycl/test/lit.cfg.py:79: note: Backend (SYCL_BE): PI_LEVEL_ZERO
llvm-lit: /WS/sycl/test/lit.cfg.py:141: warning: CPU device not found
llvm-lit: /WS/sycl/test/lit.cfg.py:159: note: Found available GPU device
llvm-lit: /WS/sycl/test/lit.cfg.py:192: warning: Accelerator device not found
llvm-lit: /WS/sycl/test/lit.cfg.py:206: note: Using opencl-aot version which is built as part of the project
llvm-lit: /WS/sycl/test/lit.cfg.py:218: note: Found pre-installed AOT device compiler ocloc
llvm-lit: /WS/sycl/test/lit.cfg.py:221: warning: Couldn't find pre-installed AOT device compiler aoc
-- Testing: 1 tests, 1 workers --
UNSUPPORTED: SYCL :: esimd/on-device/BitonicSortK.cpp (1 of 1)

Testing Time: 0.12s
  Unsupported: 1

3 warning(s) in tests
user@host:/WS/build$ bin/llvm-lit -v ../sycl/test/esimd/on-device/BitonicSortK.cpp --param SYCL_BE=PI_OPENCL
llvm-lit: /WS/build/bin/../../llvm/utils/lit/lit/llvm/config.py:347: note: using clang: /WS/build/bin/clang
llvm-lit: /WS/sycl/test/lit.cfg.py:79: note: Backend (SYCL_BE): PI_OPENCL
llvm-lit: /WS/sycl/test/lit.cfg.py:133: note: Found available CPU device
llvm-lit: /WS/sycl/test/lit.cfg.py:159: note: Found available GPU device
llvm-lit: /WS/sycl/test/lit.cfg.py:187: note: Found available accelerator device
llvm-lit: /WS/sycl/test/lit.cfg.py:206: note: Using opencl-aot version which is built as part of the project
llvm-lit: /WS/sycl/test/lit.cfg.py:218: note: Found pre-installed AOT device compiler ocloc
llvm-lit: /WS/sycl/test/lit.cfg.py:221: warning: Couldn't find pre-installed AOT device compiler aoc
-- Testing: 1 tests, 1 workers --
PASS: SYCL :: esimd/on-device/BitonicSortK.cpp (1 of 1)

Testing Time: 6.92s
  Passed: 1

1 warning(s) in tests
user@host:$ head -11 ../sycl/test/esimd/on-device/BitonicSortK.cpp
//==---------------- BitonicSortK.cpp  - DPC++ ESIMD on-device test --------==//
//
// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
//
//===----------------------------------------------------------------------===//
// TODO enable on Windows
// REQUIRES: linux && gpu
// UNSUPPORTED: cuda || level_zero
// RUN: %clangxx-esimd -fsycl %s -o %t.out
```

Reply:
Do you have Level Zero installed on this machine?
Do you mean that `SYCL_BE` impacts what devices are detected by lit? I didn't expect it to work this way, but if so, I guess original patch will also work:

// REQUIRES: linux && gpu && opencl

@vladimirlaz, could you confirm that, please?

Reply:
you can see in the logs above that L0 RT is available.

The following line will work. to run test only on linux gpu opencl runtime
// REQUIRES: linux && gpu && opencl

Reply:
changed to // REQUIRES: linux && gpu && opencl

Reply:
Thanks.
I looked though the code and confirmed what Vladimir said. 
`lit` tool ignores other devices/APIs when `--param SYCL_BE=X` is set.

Reply:
BTW, it's still unclear why we don't see any issues with these tests today in CI?
@vladimirlaz, any ideas?